compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -768,7 +771,9 @@ impl Diagnostic {
         )
     }
 
-    /// [`Diagnostic::span_suggestions()`] but you can set the [`SuggestionStyle`].
+    /// [`Diagnostic::span_suggestions()`] but you can set the [`SuggestionStyle`]. This version
+    /// *doesn't* sort the suggestions, so the caller has control of the order in which they are
+    /// presented.
```

Comment:
This seems somewhat arbitrary, why this function specifically?

Reply:
Maybe a separate function would be better? (so that you wouldn't need to change callsites, etc)

Reply:
Yeah, considered adding a new one, but I'm sad at the proliferation of different methods to do the same thing except for one difference :-/

Reply:
I wonder if we should stop sorting altogether, and rely on the caller to chose the proper order.

Reply:
The sorting was included originally because some suggestions weren't deterministic. There are only 28 uses of span_suggestions so we can probably fix them by hand. Let me see what that commit will look like.